Modules/core/Controller/CoreconnectionController.php

```diff
@@ -110,49 +108,63 @@ public function loginAction() {
             }
 
-            if ($login == "--") {
-                $this->render(array('msgError' => 'Login not correct', "admin_email" => $admin_email), "index");
-                return;
-            }
-
             $connect = $this->connect($login, $pwd);
-            //print_r($connect);
-
-            if ($connect == "allowed") {
-
+            if ($connect === "allowed") {                
                 $loggedUser = $this->initSession($login);
-
                 // generate the remember me cookie
                 if ($this->request->isparameter("remember")) {
-                    //throw new Exception("Set cookie <br/>");
                     $key = sha1($this->generateRandomKey());
-                    //echo "set cookie with id = " . $key . "<br/>";
                     $cookieSet = setcookie("auth", $loggedUser['idUser'] . "-" . $key, time() + 3600 * 24 * 3);
                     if (!$cookieSet) {
-                        //die("die failed to set cookie with key " . $key . "<br/>");
                         throw new PfmException("failed to set cookie with key " . $key);
                     }
                     $modelUser = new CoreUser();
                     $modelUser->setRememberKey($loggedUser['idUser'], $key);
                 }
-
                 if($redirection) {
                     $this->redirect($redirection);
                     return;
                 }
-
                 // redirect
                 $redirectPath = $this->getRedirectPath();
                 $this->redirectNoRemoveHeader($redirectPath);
             } else {
-                $this->redirect('/coreconnection?redirect_url='.$redirection);
-                //$this->indexAction($connect);
+                $this->loginError($redirection, $connect);
             }
         } else {
-            throw new PfmAuthException("Action not allowed : login or password undefined", 401);
+            throw new PfmAuthException(CoreTranslator::UndefinedCredentials($lang), 401);
         }
     }
 
+    /**
+     * In case of connection failure,
+     * redirects and sets flash message depending on connection error message
+     * 
+     * @param string $redirection redirection string
+     * @param string $connection_error error returned at connection failure
+     * 
+     */
+    private function loginError($redirection, $connection_error = "") {
+        $lang = $this->getLanguage();
+        $_SESSION['flashClass'] = "danger";
+        switch ($connection_error) {
+            case "inactive":
+                $msg = CoreTranslator::AccountInactive($lang);
+                break;
+            case "invalid_password":
+                $msg = CoreTranslator::InvalidPassword($lang);
+                break;
+            case "invalid_login":
+                $msg = CoreTranslator::InvalidLogin($lang);
+                break;
+            default:
+                $msg = $connection_error;

```

Comment:
default is empty.... should at least display a "something went wrong" or alike

Reply:
ah yes. At first $connection_error parameter was not optional. I add that